src/redux/auth-reducer.ts

```diff
@@ -0,0 +1,98 @@
+import { stopSubmit } from "redux-form";
+import { authAPI, securityAPI } from "../api/api";
+
+const SET_USER_DATA = 'SET-USER-DATA';
+const GET_CAPTCHA_URL_SUCCESS = 'GET-CAPTCHA-URL-SUCCESS';
+
+// export type InitialStateType = {
+//   id: number | null
+//   email: string | null
+//   login: string | null
+//   isAuth: boolean
+//   isFetching: boolean
+//   captchaUrl: string | null
+// }
+
+let initialState = {
+  id: null as number | null,
+  email: null as string | null,
+  login: null as string | null,
+  isAuth: false,
+  isFetching: false,
+  captchaUrl: null as string | null // если null, значит капча не требуется
+};
```

Comment:
Use `InitialStateType`

Reply:
on the lines [16-23] an alternative to the lines [7-14] for understanding